torch/_prims_common/__init__.py

```diff
@@ -921,7 +921,31 @@ def infer_size(shape: ShapeType, numel: int) -> Tuple[int, ...]:
         # Convert to list to produce a compatible error message with core
         # PyTorch, which prints sequences in square brackets.
         shape = list(shape)
+
+        from torch.fx.experimental.sym_node import SymNode
+        from torch.utils._sympy.functions import CleanDiv
+
         shape[dim] = numel // newsize
+        if isinstance(newsize, torch.SymInt):
+            # we now know that every component of the newsize multiplication must be >= 1
+            if newsize.node.shape_env is not None:
+                newsize.node.shape_env._constrain_range_for_size(
+                    newsize.node.expr, min=1, max=None
+                )
+                for s in newsize.node.expr.args:
+                    newsize.node.shape_env._constrain_range_for_size(s, min=1, max=None)
+
+            # Specify that this is a CleanDiv and not just a FloorDiv
```

Comment:
This is also not the right way to do this. I think what you should do is assert exactly divisibility (torch._check(numel % newsize == 0)) and then the division should automatically convert into a CleanDiv, although I haven't tried this recently so I don't know if it will work out of the box.